caddyhttp/gzip/gzip.go

```diff
@@ -161,7 +162,8 @@ func (w *gzipResponseWriter) CloseNotify() <-chan bool {
 	if cn, ok := w.ResponseWriter.(http.CloseNotifier); ok {
 		return cn.CloseNotify()
 	}
-	panic(httpserver.NonCloseNotifierError{Underlying: w.ResponseWriter})
+	log.Printf("[WARNING] gzipResponseWriter: %s\n", httpserver.NonCloseNotifierError{Underlying: w.ResponseWriter})
+	return nil
```

Comment:
This panic is triggered during `go test`: 

```
panic: *httptest.ResponseRecorder is not a closeNotifier [recovered]
        panic: *httptest.ResponseRecorder is not a closeNotifier

goroutine 54 [running]:
testing.tRunner.func1(0xc4200c8700)
        /home/tw/goroot/src/testing/testing.go:710 +0x256
panic(0x950020, 0xc42015c3e0)
        /home/tw/goroot/src/runtime/panic.go:489 +0x25b
github.com/mholt/caddy/caddyhttp/httpserver.(*ResponseRecorder).CloseNotify(0xc4201502d0, 0xc991e0)
        /home/tw/golib/src/github.com/mholt/caddy/caddyhttp/httpserver/recorder.go:97 +0xdb
github.com/mholt/caddy/caddyhttp/proxy.createUpstreamRequest(0xc991e0, 0xc4201502d0, 0xc42000b400, 0x0, 0xc42015a7e0)
        /home/tw/golib/src/github.com/mholt/caddy/caddyhttp/proxy/proxy.go:281 +0x610
github.com/mholt/caddy/caddyhttp/proxy.Proxy.ServeHTTP(0xc94560, 0x9ed2e8, 0xc42016e770, 0x1, 0x1, 0xc991e0, 0xc4201502d0, 0xc42000b400, 0x0, 0x0, ...)
        /home/tw/golib/src/github.com/mholt/caddy/caddyhttp/proxy/proxy.go:107 +0x1e2
github.com/mholt/caddy/caddyhttp/proxy.TestReverseProxy(0xc4200c8700)
        /home/tw/golib/src/github.com/mholt/caddy/caddyhttp/proxy/proxy_test.go:107 +0x898
testing.tRunner(0xc4200c8700, 0x9ed488)
        /home/tw/goroot/src/testing/testing.go:745 +0xb5
created by testing.(*T).Run
        /home/tw/goroot/src/testing/testing.go:787 +0x2d8

```

If someone isn't a `closeNotifier`, we just return a nil channel and it will block forever when waiting. I think it's okay in this case.